src/hugr/views/sibling_subgraph.rs

```diff
@@ -184,63 +181,13 @@ impl<'g, Base: HugrView> SiblingSubgraph<'g, Base> {
                 .map(to_pg),
         );
         let nodes = subpg.nodes_iter().map_into().collect_vec();
-
-        validate_subgraph(base, &nodes, &inputs, &outputs)?;
+        validate_subgraph(hugr, &nodes, &inputs, &outputs)?;
 
         if !subpg.is_convex_with_checker(&mut checker.0) {
             return Err(InvalidSubgraph::NotConvex);
         }
 
         Ok(Self {
-            base,
-            nodes,
-            inputs,
-            outputs,
-        })
-    }
-
-    /// Create a new convex sibling subgraph from a set of nodes.
-    ///
-    /// This fails if the set of nodes is not convex, nodes do not share a
-    /// common parent or the subgraph is empty.
-    pub fn try_new(
```

Comment:
Not really, the convexity check will take as much time to do its thing anyways, so recomputing the node set comes pretty much for free (other than the memory allocation).

The reason this existed was because before storing the input and output boundaries explicitly in the `struct`, the two constructors were quite different. Once both constructors needed the boundary signature as part of the arguments it became quite redudant to have both.